WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/detail/CategoryDetailViewModel.kt

```diff
@@ -1,19 +1,54 @@
 package org.wordpress.android.ui.prefs.categories.detail
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import kotlinx.coroutines.CoroutineDispatcher
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode.MAIN
+import org.wordpress.android.R
+import org.wordpress.android.R.string
 import org.wordpress.android.fluxc.Dispatcher
 import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.TaxonomyStore.OnTermUploaded
+import org.wordpress.android.models.CategoryNode
 import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.ui.mysite.SelectedSiteRepository
+import org.wordpress.android.ui.posts.AddCategoryUseCase
+import org.wordpress.android.ui.posts.GetCategoriesUseCase
+import org.wordpress.android.ui.prefs.categories.detail.CategoryUpdateUiState.Failure
+import org.wordpress.android.ui.prefs.categories.detail.CategoryUpdateUiState.InProgress
+import org.wordpress.android.ui.prefs.categories.detail.CategoryUpdateUiState.Success
+import org.wordpress.android.ui.prefs.categories.detail.SubmitButtonUiState.SubmitButtonDisabledUiState
+import org.wordpress.android.ui.prefs.categories.detail.SubmitButtonUiState.SubmitButtonEnabledUiState
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+import org.wordpress.android.util.NetworkUtilsWrapper
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ResourceProvider
 import org.wordpress.android.viewmodel.ScopedViewModel
 import javax.inject.Inject
 import javax.inject.Named
 
 class CategoryDetailViewModel @Inject constructor(
     @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
-    private val dispatcher: Dispatcher
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val getCategoriesUseCase: GetCategoriesUseCase,
+    private val addCategoryUseCase: AddCategoryUseCase,
+    resourceProvider: ResourceProvider,
+    private val dispatcher: Dispatcher,
+    selectedSiteRepository: SelectedSiteRepository
 ) : ScopedViewModel(bgDispatcher) {
     private var isStarted = false
-    lateinit var siteModel: SiteModel
+    private val siteModel: SiteModel = requireNotNull(selectedSiteRepository.getSelectedSite())
+
+    private val topLevelCategory = CategoryNode(0, 0, resourceProvider.getString(R.string.top_level_category_name))
+
+    private val _onCategoryPush = MutableLiveData<Event<CategoryUpdateUiState>>()
+    val onCategoryPush: LiveData<Event<CategoryUpdateUiState>> = _onCategoryPush
```

Comment:
🔍 We're using two different terms `push` and `add` for the same action. We should try to use consistent naming for same action.

Reply:
@ashiagr, I added this with the intention that the same live data will be used in the UI when we are updating a category(Edit/Delete). I can see there is a consistency issue in the naming. I can rename the live data to `addCategoryState` or something else to be consistent with the add category implementation. When we implement the Edit/Delete we can rename this to something appropriate?. Wdyt ?

Reply:
Makes sense, thanks for the clarification. We can wait till we add Edit/Delete implementation.